pkg/controller/elasticsearch/validation/node_labels_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package validation_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/validation"
+)
+
+func TestNodeLabels_IsAllowed(t *testing.T) {
+	type args struct {
+		nodeLabel string
+	}
+	tests := []struct {
+		name              string
+		exposedNodeLabels []string
+		args              args
+		want              bool
+		wantErr           bool
+	}{
+		{
+			name:              "Cannot parse label",
+			exposedNodeLabels: []string{"*"},
```

Comment:
".*" would match any label, should we test that as well? And maybe one test where the `exposedNodeLabels` do not contain a wildcard? I know that's borderline testing Go's regex engine but maybe for completeness?